### PR TITLE
schema.sql: Replace `varchar` with `text`

### PR DESCRIPTION
**Pull request by KellerFuchs**

### CODE REVIEW

**Comment by daurnimator** [comment]
Sure.... why not I guess.

**Review by daurnimator — APPROVED** [review]

File: schema.sql [hunk 1]
```diff
@@ -18,8 +18,8 @@ CREATE TABLE "hosts" (
 --  https://www.debian.org/doc/debian-policy/ch-opersys.html#s9.2.2
 CREATE SEQUENCE user_id MINVALUE 4000 MAXVALUE 59999 NO CYCLE;
 
-CREATE DOMAIN username_t varchar(31) CHECK (
```
Comment (mayli) [inline schema.sql:21]: using `text` means no index on this field, is this intended?

Comment (mayli) [inline schema.sql:21]: the `getpwnam` is querying by the username.
`getpwnam        = SELECT username, passwd, gecos, homedir, shell, uid, gid FROM passwd_table WHERE username = $1``

Comment (KellerFuchs) [reply schema.sql:21]: Not sure where you got the idea that Postgres does not make indexes for `text` columns.
It definitely has a btree index here, due to the `UNIQUE` constraint:
```
> \d passwd
+----------+------------+------------------------------------------------+
| Column   | Type       | Modifiers                                      |
|----------+------------+------------------------------------------------|
| uid      | integer    |  not null default nextval('user_id'::regclass) |
| name     | username_t |  not null                                      |
| host     | text       |  not null                                      |
| homedir  | text       |  not null                                      |
| data     | jsonb      |                                                |
+----------+------------+------------------------------------------------+
Indexes:
    "passwd_pkey" PRIMARY KEY, btree (uid)
    "passwd_name_key" UNIQUE CONSTRAINT, btree (name)
Check constraints:
    "data_user" CHECK (validate_json_schema('{"type": "object", "title": "#! userdb -- Schema for user records au
Foreign-key constraints:
    "passwd_host_fkey" FOREIGN KEY (host) REFERENCES hosts(name)
Referenced by:
    TABLE "aux_groups" CONSTRAINT "aux_groups_uid_fkey" FOREIGN KEY (uid) REFERENCES passwd(uid) ON DELETE CASCAD
Triggers:
    max_users AFTER INSERT OR UPDATE ON passwd NOT DEFERRABLE INITIALLY IMMEDIATE FOR EACH ROW EXECUTE PROCEDURE
```

Comment (KellerFuchs) [reply schema.sql:21]: Yes, and?

Comment (mayli) [reply schema.sql:21]: nvm, that's limitation in MySQL

**Pull request merged**